backend/services/implementations/statisticService.ts

```diff
@@ -112,6 +112,42 @@ class StatisticService implements IStatisticService {
     return this.constructTestStatisticsByGroup(aggCursor, "school");
   }
 
+  /* eslint-disable class-methods-use-this */
+  async getSubmissionCountByTest(testId: string): Promise<number> {
+    const pipeline = [
+      // Stage 1: filter out tests that have the requested testId
+      { $match: { test: { $eq: Types.ObjectId(testId) } } },
```

Comment:
can you help refactor each of these queries into constants (or functions if they take in params). you can make a new file called like `pipelineQueryUtils.ts` in `/utilities` to put them in.

e.g. 
```
export const filterTestsByTestIdQuery = (testId: string) => {
  return { 
    $match: { test: { $eq: Types.ObjectId(testId) } } 
  }
}
```
```suggestion
      filterTestsByTestIdQuery(testId),
```

Reply:
This seems like a good idea, will do!